q2_fmt/_peds.py

```diff
@@ -466,11 +479,458 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+    number of mismatched PEDS value that are greater than the actual p-value.
+    p-values are corrected by adding 1 to each side of the faction
+    because Monte Carlo Simulation is able to produce an invalid p-value of 0.
+
+    Parameters
+    ----------
+    mismatched_peds: pd.DataFrame
+        A Dataframe that contains all mismatched PEDS values.
+    actual_temp: pd.Series
+        A Series containing Sample IDs as the index and actual PEDS values for
+        the sample as the value.
+    iterations: int
+        Number of iterations to run simulations (Number of times to randomly
+        sample mismatched_peds)
+    mismatched_peds_n: int
+        Number of mismatched donor pairs are possible.
+
+    Returns
+    -------
+    per_subject_stats: pd.DataFrame
+        a Dataframe containing test statistics, p-values, q-values and
+        sample size info. Note: this is the DataFrame that is returned
+        to the user as the per-subject-stats.qza
+
+    Examples
+    --------
+    mismatched_peds = [0, 0, 0, 0]
+    actual_temp = pd.Series([data = [1, 1, 1, 1],
+               index = [sample1, sample2, sample3, sample4]])
+    iterations = 10
+    mismatched_pairs_n = 4
+
+    _per_subject_stats(mismatched_peds, actual_temp,
+                       iterations, mismatched_pairs_n):
+
+    pd.Dataframe({"A:group": [sample1, sample2, sample3, sample4],
+                   "A:n": [1, 1, 1, 1],
+                   "A:measure" : [1, 1, 1, 1],
+                   "B:group" : [shuffled_recipients, shuffled_recipients,
+                                shuffled_recipients, shuffled_recipients],
+                   "B:n": [4, 4, 4, 4],
+                   "B:measure": [0, 0, 0, 0],
+                   "n": [10, 10, 10, 10],
+                   "test-statistic": [10, 10, 10, 10],
+                   "p-value": [0.001, 0.001, 0.001, 0.001],
+                   "q-value": [0.004, 0.002, 0.00133, 0.001]})
+    """
+    disagree_df = pd.DataFrame([])
+    peds_iter_mean = []
+    for value in actual_temp:
+        peds_iters = _simulate_uniform_distro(mismatched_peds, iterations)
+        peds_iter_mean.append(peds_iters.mean())
+        disagree_recipient = peds_iters >= value
+        # add disagree recipient as a row in the disagree df
+        disagree_df = pd.concat([disagree_df, disagree_recipient.to_frame().T])
+    disagree_series = disagree_df.sum(axis=1)
+    agree_series = iterations-disagree_series
+    # adding 1 here because you can mathmatically can get p-value of 0 from a
+    # Monte Carlo Simulation
+    per_subject_p = (disagree_series + 1)/(iterations+1)
+    per_subject_q = false_discovery_control(ps=per_subject_p, method='bh')
+    per_sub_stats = pd.DataFrame({'A:group': actual_temp.index,
+                                 'A:n': 1,
+                                  'A:measure': actual_temp.values,
+                                  'B:group': "shuffled_recipients",
```

Comment:
I'd remove the `_`, if this is just a value that is being presented in a table to a user.